discovery_engine_core/tooling/bin/clean_sources.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    env::args,
+    fs::{remove_file, File},
+    io::{BufRead, BufReader, BufWriter, Write},
+    path::Path,
+};
+
+use anyhow::{bail, Result};
+use csv::{QuoteStyle, ReaderBuilder, WriterBuilder};
+
+fn main() -> Result<()> {
+    let in_file = args().nth(1).unwrap();
+    let tmp_file = Path::new(&in_file).with_extension("tmp");
+    let out_file = args().nth(2).unwrap();
+
+    fix_sources(in_file, &tmp_file)?;
+    clean_sources(&tmp_file, &out_file)?;
+    check_sources(out_file)?;
+    remove_file(tmp_file)?;
+
+    Ok(())
+}
+
+/// Fixes illegal linebreaks in the sources csv file.
+fn fix_sources(in_file: impl AsRef<Path>, out_file: impl AsRef<Path>) -> Result<()> {
+    let reader = BufReader::new(File::open(in_file)?);
+    let mut writer = BufWriter::new(File::create(out_file)?);
+
+    let mut broken_line = String::new();
+    for line in reader.lines() {
+        let line = line?;
+        if !broken_line.is_empty() || line.split(';').count() != 4 {
+            if broken_line.is_empty() {
+                broken_line = line;
+            } else {
+                broken_line.push(' ');
+                broken_line.push_str(&line);
+            }
+            if broken_line.split(';').count() == 4 {
+                writer.write_all(broken_line.as_bytes())?;
+                writer.write_all(&[b'\n'])?;
+                broken_line.clear();
+            }
```

Comment:
❔I'm not sure if it makes sense to return an error/panic, if we for some reason get `broken_line.split(';').count() > 4` wdyt?

Reply:
in general we should get wellformed csv files, even the broken lines shouldn't be there, but unfortunately it's not up to us to ensure this. for now i would just keep the script as is, if we need to we can always update it later on.